docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,640 @@
+
+In `HttpWebRequest`, you might have used the `CachePolicy` property to set these headers. However, in `HttpClient`, you need to manually set these headers on the request.
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+request.CachePolicy = new HttpRequestCachePolicy(HttpRequestCacheLevel.NoCacheNoStore);
+using HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+```
+
+In the older `HttpWebRequest` API, applying `CachePolicy` was straightforward due to its built-in support for this feature. However, the newer `HttpClient` API does not provide the same built-in functionality. Despite this, you can achieve similar behavior by implementing a `AddCacheControlHeaders` that manually add cache related headers.
+
+**New Code Using `HttpClient`**:
+
+:::code source="../snippets/httpclient/AddCacheControlHeaders.cs" id="CachePolicy":::
+
+You can find implementation of `AddCacheControlHeaders` [here](../snippets/httpclient/AddCacheControlHeaders.cs).
+
+`AddCacheControlHeaders` Usage:
+:::code source="../snippets/httpclient/Program.CacheControlHeaders.cs" id="CacheControlProgram":::
+
+## Usage of Buffering Properties
+
+When migrating from HttpWebRequest to `HttpClient`, it's important to understand the differences in how these two APIs handle buffering.
+
+**Old Code Using `HttpWebRequest`**:
+
+In `HttpWebRequest`, you have direct control over buffering properties through the `AllowWriteStreamBuffering` and `AllowReadStreamBuffering` properties. These properties enable or disable buffering of data sent to and received from the server.
+
+```csharp
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+request.AllowReadStreamBuffering = true; // Default is `false`.
+request.AllowWriteStreamBuffering = false; // Default is `true`.
+```
+
+**New Code Using `HttpClient`**:
+
+In `HttpClient`, there are no direct equivalents to the `AllowWriteStreamBuffering` and `AllowReadStreamBuffering` properties.
+
+However, you can achieve similar outcome of `disable write buffering` behavior by using the `StreamContent` class for streaming data. Here's an example:
+
+```csharp
+HttpClient client = new HttpClient();
+
+HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, uri);
+request.Content = new StreamContent(yourStream);
+request.Headers.TransferEncodingChunked = true;
+
+HttpResponseMessage response = await client.SendAsync(request);
+```
+
+In `HttpClient` read buffering is enabled by default. `Disabling read buffering` is much easier. Here's an example:
+
+```csharp
+HttpClient client = new HttpClient();
+
+HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
```

Comment:
```suggestion
In `HttpClient` read buffering is enabled by default. To avoid it, you may specify the `HttpCompletionOption.ResponseHeadersRead` flag, or use the `GetStreamAsync` helper.

```csharp
HttpClient client = new HttpClient();

using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
await using Stream responseStream = await response.Content.ReadAsStreamAsync();

// Or simply
await using Stream responseStream = await client.GetStreamAsync(uri);
```